torch/csrc/jit/serialization/import_source.cpp

```diff
@@ -475,6 +478,69 @@ struct SourceImporterImpl : public Resolver,
     cu_->define(qualified_classname, methods, resolvers, &self);
   }
 
+  void importEnum(
+      const QualifiedName& qualified_name,
+      const ClassDef& enum_def) {
+    std::vector<std::pair<std::string, IValue>> names_values;
+
+    TypePtr value_type = nullptr;
+    auto set_or_check_type = [&value_type](
+                                 const TypePtr& t, const SourceRange& loc) {
+      if (!value_type) {
+        value_type = t;
+      } else if (value_type != t) {
+        throw ErrorReport(loc)
+            << "Enum class with varying value types are not supported.";
+      }
+    };
+
+    for (const auto& statement : enum_def.body()) {
+      if (statement.kind() != TK_ASSIGN) {
+        throw ErrorReport(statement.range())
+            << "Unexpected statement in Enum class body: "
+               "only enum attribute definitions are currently supported.";
+      }
+
+      const auto assign = Assign(statement);
+      auto name = Var(assign.lhs()).name().name();
+
+      IValue ivalue;
+      auto rhs = assign.rhs().get();
+      switch (rhs.kind()) {
+        case TK_STRINGLITERAL:
+          ivalue = IValue(StringLiteral(rhs).text());
+          set_or_check_type(StringType::get(), statement.range());
+          break;
+        case TK_CONST: {
+          auto numeric_const = Const(rhs);
+          if (numeric_const.isFloatingPoint()) {
+            ivalue = IValue(numeric_const.asFloatingPoint());
+            set_or_check_type(FloatType::get(), statement.range());
+          } else if (numeric_const.isIntegral()) {
+            ivalue = IValue(numeric_const.asIntegral());
+            set_or_check_type(IntType::get(), statement.range());
+          }
+          break;
+        }
+        default:
+          throw ErrorReport(rhs.range())
+              << "Unsupported enum value type: " << rhs.kind()
+              << ". Only Integers, Floats and Strings are supported.";
+      }
+
+      names_values.emplace_back(std::make_pair(name, ivalue));
+    }
+
+    if (!value_type) {
+      throw ErrorReport(enum_def.range())
+          << "No enum values defined for " << qualified_name.qualifiedName();
+    }
+
```

Comment:
Consider checking also that there is at least one item in `names_values`. Although that *must* be true if `value_type` is not `nullptr`...

Reply:
Yep, that's why value_type is being checked.